add beforeRetryMethod parameter to RetryOnFailure

agdula opened the pull request:
I needed a method that would be invoked before another try of running integration test.

Use Case:
1. your test fails from time to time but when it fails remaining tests will fail too
2. before you retry you need to have environment in a certain condition ( restart services, restart OS for Selenium IEDriver ... whatever )
3. The step 2 is quicker than re running whole suite again

Notice:
The change slightly changes ( I would say fixes ) using default value of retry from System.properties  ( test adjusted )

agdula commented:
I've resolved conflicts so that you have less work with this request

anotherchrisberry commented:
Is there any reason to require the name of the method be declared? This seems like a feature-type method (e.g. `setup`, `cleanup`, `setupSpec`) that would be run if present based on its presence in the specification file. That would simplify the configuration, but maybe I'm missing something.

agdula commented:
Good point. For my case there is no good reason for introducing method name as a parameter. I think that the risk of method name collision is low so it could be static. I'll simplify the approach. Thx

Commit:
add beforeRetry method to methods invoked

I needed a method that would be invoked before another retrying of running integration test that is not one of setup, cleanup or setupSpec.

Use Case:
1. your test fails from time to time but when it fails remaining tests will fail too
2. before you retry you need to have environment in a certain condition ( restart services, restart OS for Selenium IEDriver ... whatever )
3. The step 2 is quicker than re running whole suite again

agdula commented:
I've force pushed a solution with a simpler approach. Cheers!

anotherchrisberry commented:
LGTM - thanks for including a test!
